src/Controls/tests/Core.UnitTests/ShellTests.cs

```diff
@@ -255,6 +256,36 @@ public void CurrentItemDoesNotChangeOnSecondAdd()
 			Assert.Equal(shellItem, shell.CurrentItem);
 		}
 
+		[Fact]
+		public void AddRemoveItemsDoesNotCrash()
```

Comment:
If I'm following the code in this PR correctly, I don't think this UnitTest adds any coverage to the added code. 

Both code chunks added will only execute when they are running as a WinUI app. 

Can we add the repro [here](https://github.com/dotnet/maui/issues/9938) to a device test and then work from there?

Reply:
Just got back from vacation -- for sure will add a device test.